src/pages/Home/ui/Calendar/ui/CalendarModal/calendarModal.hook.ts

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+import {
+  CalendarEvent,
+  ScheduleCategory,
+  TodoPriority,
+  Schedule,
+  Todo,
+} from 'entities/calendar/type';
+
+interface UseEventStateProps {
+  event?: CalendarEvent;
+}
+
+const useEventState = ({ event }: UseEventStateProps) => {
+  const [eventType, setEventType] = useState<'Schedule' | 'Todo'>(
+    event?.type || 'Schedule'
+  );
+  const [title, setTitle] = useState(event?.title || '');
+  const [content, setContent] = useState(event?.content || '');
+  const [selectedRepeat, setSelectedRepeat] = useState<string>(
+    event?.repeatCycle || 'NONE'
+  );
+  const [selectedCategory, setSelectedCategory] = useState<ScheduleCategory>(
+    (event?.category as ScheduleCategory) || 'FIRST'
+  );
+  const [selectedPriority, setSelectedPriority] = useState<TodoPriority>(
+    (event?.priority as TodoPriority) || 'MIDDLE'
+  );
+  const [isAllDay, setIsAllDay] = useState(
+    eventType === 'Schedule' ? (event as Schedule)?.allDay || false : false
+  );
+  const [location, setLocation] = useState(event?.location || '');
+
+  const isInitial = !event || event.title === '';
+
+  useEffect(() => {
+    if (event) {
+      setEventType(event.type);
+      setTitle(event.title);
+      setContent(event.content || '');
+      setSelectedRepeat(event.repeatCycle || 'NONE');
+      setLocation(event.location || '');
+
+      if (event.type === 'Schedule') {
+        setSelectedCategory((event as Schedule).category);
+        setIsAllDay((event as Schedule).allDay);
+      } else {
+        setSelectedPriority((event as Todo).priority);
+      }
+    }
+  }, [event]);
```

Comment:
```suggestion
  useEffect(() => {
    if (event) {
      setState({
        eventType: event.type,
        title: event.title,
        content: event.content || '',
        selectedRepeat: (event.repeatCycle || 'NONE') as RepeatValue,
        selectedCategory: event.type === 'Schedule' ? (event as Schedule).category : undefined,
        selectedPriority: event.type === 'Todo' ? (event as Todo).priority : undefined,
        isAllDay: event.type === 'Schedule' ? (event as Schedule).allDay || false : false,
        location: event.location || '',
      });
    }
  }, [event]);
```

위와 같이 묶고 나면 이런식으로 간소화 가능할 거 같네요!